connection.go

```diff
@@ -948,6 +1033,51 @@ func (conn *Connection) timeouts() {
 	}
 }
 
+func (conn *Connection) contexts() {
+	timeout := time.Nanosecond * 1
```

Comment:
Why not 5 ns, 10 ns, 100 ns, 1 us, 1 ms? Once you start using timings, these questions arise.

The idea is clear, it will work. The problem is that this won't always work:

Imagine that you have a slow CPU core and a lot of requests. If checking all requests starts to take > 1ns, then you will start checking them all the time, without stopping on the timer. You will end up consuming about 100% of a CPU core just on the requests checking.

I did not reproduce this example, perhaps in practice the behavior will be different. What I'm trying to say is that such timings lead to unexpected problems.

Reply:
Of course, I was not sure about those timings. One of the reasons to leave the context inside the Future object, was to avoid the need to check context only with this goroutine every nanosecond(or some other time). I would like to have this goroutine as a garbage collector just for cleaning up the queue when it is needed. I also had an idea to have an internal atomic counter of a Futures count inside and run this goroutine when this count is critical (we could configure this critical number and have a dependency from GOMAXPROC by default).

Reply:
It sounds like a premature optimization.